controllers/auth.js

```diff
@@ -1,29 +1,67 @@
+if(process.env.NODE_ENV!=="production"){
+  require('dotenv').config()
+}
+const db_url=process.env.DB_URI
+
+const mongoose = require('mongoose');
+var User = require("../models/user");
+var passport = require("passport");
+mongoose.connect(db_url, { useNewUrlParser: true, useUnifiedTopology: true })
+  .then(() => {
+    console.log("MONGO CONNECTION OPEN!!!")
+  })
+  .catch(err => {
+    console.log("OH NO MONGO CONNECTION ERROR!!!!")
+    console.log(err)
+  })
+
+
 var getLogin = (req, res) => {
   //TODO: render login page
+  res.render("login", { title: "Login" });
 };
 
 var postLogin = (req, res) => {
   // TODO: authenticate using passport
   //On successful authentication, redirect to next page
+  passport.authenticate("local", function (err, user, info) {
+    if (!user) {
+
+      return res.redirect("/login");
```

Comment:
Show a valid Error message here (You could use Flash for this), to tell the user that login was unsuccessful or the username/ password is incorrect.